pandas/core/categorical.py

```diff
@@ -1956,3 +1957,46 @@ def _convert_to_list_like(list_like):
     else:
         # is this reached?
         return [list_like]
+
+
+def _factorize(values):
```

Comment:
per my comment below, I think we should name this:

``_from_array``
``_from_iterables``

I think maybe more clear.